lib/perl/Genome/InstrumentData/AlignmentResult/Command/RecreatePerLaneBam.pm

```diff
@@ -156,14 +168,17 @@ sub _revert_markdup {
 
 
 sub _reheader_bam {
-    my ($self, $temp_bam) = @_;
-
-    my @header_files = glob($self->_output_dir ."/*.header");
-    unless (@header_files and @header_files == 1) {
-        die $self->error_message("Failed to get 1 header file from original per lane bam location");
+    my ($self, $temp_bam, $bam_header) = @_;
+
+    unless ($bam_header) { #find bam header from planned places
```

Comment:
If this is kept it would probably be clearer to extract this heuristic as a `_default_bam_header()` method or something.